### PR TITLE
Feature/user controller

### PR DESCRIPTION
**Pull request by Zablas**
Add User controller with GET, POST and PUT request implementations.

### CODE REVIEW

File: Backend/Controllers/UsersController.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        [HttpPut("{id}")]
+        public async Task<ActionResult> EditUser(int id, [FromBody] EditUserDTO dto)
+        {
+            var user = await Context.Users
+                .FirstOrDefaultAsync(u => u.Id == id);
+
+            if (user == null) return ApiNotFound("User does not exist!");
+
+            user.Name = dto.Name ?? user.Name;
```
Comment (wizacass) [inline Backend/Controllers/UsersController.cs:59]: Maybe we could extract this logic to model?

**Review by wizacass — CHANGES_REQUESTED** [review]

File: Backend/Controllers/UsersController.cs (hunk 1 above)
Comment (Zablas) [reply Backend/Controllers/UsersController.cs:59]: As in have a UserObj.UpdateProperties(EditUserDTO dto) method?

Comment (wizacass) [reply Backend/Controllers/UsersController.cs:59]: Yes. Something like `UpdateFromDTO()`

Comment (Zablas) [reply Backend/Controllers/UsersController.cs]: Moved.

**Review by wizacass — APPROVED** [review]

**Pull request merged**